framework/test/unit/modules/auth/auth_module.spec.ts

```diff
@@ -93,6 +100,213 @@ describe('AuthModule', () => {
 			});
 	});
 
+	describe('afterGenesisBlockExecute', () => {
+		const address = getRandomBytes(20);
+		const publicKey = getRandomBytes(32);
+		const validAsset = {
+			authDataSubstore: [
+				{
+					storeKey: address,
+					storeValue: {
+						numberOfSignatures: 0,
+						mandatoryKeys: [],
+						optionalKeys: [],
+						nonce: BigInt(23),
+					},
+				},
+				{
+					storeKey: getRandomBytes(20),
+					storeValue: {
+						numberOfSignatures: 3,
+						mandatoryKeys: [getRandomBytes(32), getRandomBytes(32)].sort((a, b) => a.compare(b)),
+						optionalKeys: [getRandomBytes(32), getRandomBytes(32)].sort((a, b) => a.compare(b)),
+						nonce: BigInt(1),
+					},
+				},
+			],
+		};
+		const invalidTestData = [
+			[
+				'when store key is not 20 bytes',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 0,
+								mandatoryKeys: [],
+								optionalKeys: [],
+								nonce: BigInt(1),
+							},
+						},
+					],
+				},
+			],
+			[
+				'mandatory key is not lexicographically sorted',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 3,
+								mandatoryKeys: [getRandomBytes(32), getRandomBytes(32)].sort((a, b) =>
+									b.compare(a),
+								),
+								optionalKeys: [],
+								nonce: BigInt(1),
+							},
+						},
+					],
+				},
+			],
+			[
+				'mandatory key is not unique',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 2,
+								mandatoryKeys: [publicKey, publicKey],
+								optionalKeys: [],
+								nonce: BigInt(1),
+							},
+						},
+					],
+				},
+			],
+			[
+				'optional key is not lexicographically sorted',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 3,
+								mandatoryKeys: [],
+								optionalKeys: [getRandomBytes(32), getRandomBytes(32)].sort((a, b) => b.compare(a)),
+								nonce: BigInt(1),
+							},
+						},
+					],
+				},
+			],
+			[
+				'optional key is not unique',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 2,
+								mandatoryKeys: [],
+								optionalKeys: [publicKey, publicKey],
+								nonce: BigInt(1),
+							},
+						},
+					],
+				},
+			],
+			[
+				'exceed total keys',
+				{
+					authDataSubstore: [
+						{
+							storeKey: getRandomBytes(8),
+							storeValue: {
+								numberOfSignatures: 36,
+								mandatoryKeys: new Array(33)
```

Comment:
Ishan's `Array.from({ length: 33 }, _ => getRandomBytes(32))` trick is really neat, you might consider it